pingpong/api_keys.py

```
@@ -155,32 +155,55 @@ async def get_process_redacted_project_api_keys(
+    )
+
+
+async def set_as_default_api_key(
+    session: AsyncSession,
+    redacted_key: str,
+    key_name: str,
+    provider: str,
+    endpoint: str | None = None,
+) -> None:
+    if provider == "azure" and endpoint is None:
+        raise ValueError("Azure endpoint required for Azure API key")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)